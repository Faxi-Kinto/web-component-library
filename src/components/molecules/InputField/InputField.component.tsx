import React from 'react';
import * as Styled from './InputField.styles';
import { FieldProps } from 'components/molecules/FormField/FieldProps';
import Input, { InputProps } from 'components/atoms/Input/Input.component';
import useField from 'hooks/useField';

/**
 * @name InputField
 *
 * [Insert omponent description]
 *
 * @returns {JSX}
 */
type ValueOnChange = (event: React.ChangeEvent<HTMLInputElement>) => void;

export type InputFieldProps = FieldProps<string, ValueOnChange> & {
  description?: string;
} & InputProps;

const InputField: React.FC<InputFieldProps> = (
  props: InputFieldProps
): JSX.Element => {
  const { error, description, dirty, touched, ...rest } = props;

  const showError = useField(props);
  return (
    <Styled.Container>
      <Input error={showError} {...rest} />
      {description ? (
        <div className="input__description-msg">{description}</div>
      ) : (
        ''
      )}
      {showError ? <div className="input__error-msg">{error}</div> : ''}
    </Styled.Container>
  );
};

export default InputField;
