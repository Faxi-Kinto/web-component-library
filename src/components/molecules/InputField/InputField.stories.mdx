import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { withKnobs, text, boolean } from '@storybook/addon-knobs';

import InputField from "./";

<Meta title="Core|Molecules/InputField" />

# InpitField Component
This field is a wrapper around [Input](/docs/core-atoms-input--default-story) atom and it is intended for use in [Form Field](/docs/core-molecules-formfield--default-story)
enabeling error messages, error state for input atom and description message for the field.
InputField implements FormField API.

## Implementation

#### Default
<Preview>
  <Story name="default" decorators={[withKnobs]}>
    <InputField placeholder={text('Placeholder text', '')} description={text('Description text','')} error={text('Error message','This field is required')} touched={boolean('Touched',false)} dirty={boolean('Dirty', false)}/>
  </Story>
</Preview>

#### With Error message
<Preview>
  <Story name="With error message">
    <InputField error="Error message" touched={true} dirty={true} />
  </Story>
</Preview>

#### With Description message
<Preview>
  <Story name="With description message">
    <InputField description="Feild input description" />
  </Story>
</Preview>

## Props

<Props of={InputField} />
