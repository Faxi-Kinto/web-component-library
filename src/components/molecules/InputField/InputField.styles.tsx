import styled from 'styled-components';
import { theme } from 'config';
import { pxToRem } from 'styles/fonts';
import { marginChildren } from 'styles/spacings';

export const Container = styled.div`
  ${marginChildren(`0 0 ${pxToRem('10px')} 0`)};

  .input {
    &__description-msg {
      color: ${theme.pallet.DISABLED};
      line-height: ${pxToRem('20px')};
    }

    &__error-msg {
      font-size: ${pxToRem('14px')};
      color: ${theme.pallet.ERROR};
    }
  }
`;
