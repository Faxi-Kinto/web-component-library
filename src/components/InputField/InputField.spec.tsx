import React from 'react';
import renderer from 'react-test-renderer';

import InputField from './InputField.component';

test('InputField renders correctly', () => {
  const component = renderer.create(<InputField />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
