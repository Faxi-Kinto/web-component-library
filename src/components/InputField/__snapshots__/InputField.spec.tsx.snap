// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
.c0 > * {
  margin: 0 0 0.625rem 0;
}

.c0 > :last-child {
  margin: 0;
}

.c0 .input__description-msg {
  color: #95A1A6;
  line-height: 1.25rem;
}

.c0 .input__error-msg {
  font-size: 0.875rem;
  color: #EB4820;
}

.c1 {
  width: 100%;
}

.c1 > * {
  margin: 0 0 0.625rem 0;
}

.c1 > :last-child {
  margin: 0;
}

.c1 > label {
  color: #4E606A;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 24px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-bottom: 5px;
  font-size: 1rem;
  line-height: 16px;
}

.c1 > input {
  width: 100%;
  height: 2.8125rem;
  padding: 0.5rem 1rem 0.5rem 1rem;
  box-shadow: 0.03125rem 0.03125rem 0.078125rem #00000029;
  border: 0.0625rem solid #95A1A6;
  outline: none;
}

.c1 > input.input--error {
  border: 0.0625rem solid #EB4820;
}

.c1 > input:focus,
.c1 > input.input--error {
  box-shadow: none;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <input
      className=""
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
  </div>
  
  
</div>
`;
